### PR TITLE
fixes #3544 - Editing an oVirt compute resource allows changing the type, which is unsupported

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by domcleal** [comment]
This should also be enforced at the model level so users can't change it via the API.

**Comment by lzap** [comment]
I agree, this one is easy to prevent in the model.

File: app/models/compute_resource.rb [hunk 1]
```diff
@@ -26,6 +26,7 @@ class ComputeResource < ActiveRecord::Base
   has_and_belongs_to_many :users, :join_table => "user_compute_resources"
   validates :name, :presence => true, :uniqueness => true,
             :format => { :with => /\A(\S+)\Z/, :message => N_("can't contain white spaces.") }
+  validate :ensure_provider_not_changed, :on => :update
```
Comment (isratrade) [inline app/models/compute_resource.rb:29]: @lzap, @domcleal, added model validation to ensure 'provider' is not changed on update.

File: app/models/compute_resource.rb [hunk 2]
```diff
@@ -281,4 +282,10 @@ def set_attributes_hash
     self.attrs ||= {}
   end
 
+  def ensure_provider_not_changed
+    if self.type_changed?
+      errors.add(:provider,"cannot be changed")
```
Comment (lzap) [inline app/models/compute_resource.rb]: Nitpick: Missing space

**Comment by lzap** [comment]
Going to [test] this now.

**Comment by lzap** [comment]
Works both in the UI and CLI

```
[lzap@lzapx hammer-cli]$ be bin/hammer compute-resource update --id 1 --provider Ovirt
Could not update the compute resource:
  Provider cannot be changed
```

**Comment by isratrade** [comment]
@lzap, fixed per your comment and also fixed failing tests that were updating provider.

File: app/models/compute_resource.rb [hunk 3]
```diff
@@ -281,4 +282,10 @@ def set_attributes_hash
     self.attrs ||= {}
   end
 
+  def ensure_provider_not_changed
+    if self.type_changed?
+      errors.add(:provider, "cannot be changed")
```
Comment (domcleal) [inline app/models/compute_resource.rb]: extract for i18n

Comment (isratrade) [inline app/models/compute_resource.rb]: i18n issue fixed and rebased

**Comment by lzap** [comment]
Hmmm I still see failures.

**Comment by isratrade** [comment]
@lzap, fixed syntax error in test file. All should pass now.

**Comment by lzap** [comment]
Thanks, great. Merged as 7e74d71 for Foreman 1.7. Thank you @isratrade !

**Pull request closed without merging**